stack.yaml

```diff
@@ -9,6 +9,7 @@ extra-deps:
   - repline-0.2.0.0
   - serialise-0.2.1.0
   - neat-interpolation-0.3.2.4
+  - dotgen-0.4.2@sha256:309b7cc8a3593a8e48bee7b53020d5f72db156d58edf78a0214f58fbb84b292b
```

Comment:
What is the reason for selecting this version of `dotgen`?

Reply:
It's the latest one (and it was available on `nixpkgs`). Also according to the changelog it contains a fix to build with GHC 7.10: https://hackage.haskell.org/package/dotgen-0.4.2/changelog
Do you think we should relax the lower version bound?

The `stack.yaml` fix should have really gone in the previous PR, so I thought I would include it here, but it's not really the focus of this short PR. Would you want me to split it in two PRs?

Reply:
@basile-henry: It's fine if you include the change in this pull request.  I was just curious

You don't need to relax the lower version bound unless people specifically ask for it